src/sage/combinat/set_partition.py

```diff
@@ -620,7 +624,11 @@ def __init__(self, parent, s, check=True):
             {}
         """
         self._latex_options = {}
-        ClonableArray.__init__(self, parent, sorted(map(frozenset, s), key=min), check=check)
+        try:
+            s = sorted(map(frozenset, s), key=min)
+        except TypeError:
```

Comment:
comparing using min of string looks strange.. 
```
sage: min(str((3,4)))
' '
```

Reply:
We shouldn't be applying `min` to a single string here, but rather to an iterable of strings.
```
sage: min(";-)")
)
sage: min([";-)"])
';-)'
```